test/models/user_test.exs

```diff
@@ -0,0 +1,18 @@
+defmodule Yggdrasil.UserTest do
+  use Yggdrasil.ModelCase
+
+  alias Yggdrasil.User
+
+  @valid_attrs %{hash: "some content", username: "some content"}
+  @invalid_attrs %{}
+
+  test "changeset with valid attributes" do
```

Comment:
Missing a lot of test cases for validations. Is it possible to also include the unique username constraint as a test?